internal/server/server.go

```diff
@@ -0,0 +1,179 @@
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/Konstantsiy/image-converter/internal/auth"
+	"github.com/Konstantsiy/image-converter/internal/domain"
+	"github.com/Konstantsiy/image-converter/internal/repository"
+	"github.com/Konstantsiy/image-converter/internal/validation"
+	"github.com/gorilla/mux"
+)
+
+// Server represents application server.
+type Server struct {
+	repo         *repository.Repository
+	tokenManager *auth.TokenManager
+}
+
+// NewServer creates new application server.
+func NewServer(repo *repository.Repository, tokenManager *auth.TokenManager) *Server {
+	return &Server{
+		repo:         repo,
+		tokenManager: tokenManager,
+	}
+}
+
+// AuthMiddleware checks user authorization.
+func (s *Server) AuthMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// check JWT tokens
+		next.ServeHTTP(w, r)
+	})
+}
+
+// RegisterRoutes registers application routers.
+func (s *Server) RegisterRoutes(r *mux.Router) {
+	r.HandleFunc("/user/login", s.LogIn).Methods("POST")
+	r.HandleFunc("/user/signup", s.SignUp).Methods("POST")
+
+	api := r.NewRoute().Subrouter()
+	api.Use(s.AuthMiddleware)
+
+	api.HandleFunc("/conversion", s.ConvertImage).Methods("POST")
+	api.HandleFunc("/images/{id}", s.DownloadImage).Methods("GET")
+	api.HandleFunc("/requests", s.GetRequestsHistory).Methods("GET")
+}
+
+// LogIn implements the user authentication process.
+func (s *Server) LogIn(w http.ResponseWriter, r *http.Request) {
+	var request domain.AuthRequest
+
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+	defer r.Body.Close()
+
+	var user domain.UserInfo
+	user, err = s.repo.GetUserByEmail(request.Email)
+	if err == repository.ErrNoSuchUser {
+		http.Error(w, "can't get user info: "+err.Error(), http.StatusUnauthorized)
+		return
+	}
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	var tokens domain.TokensResponse
+	tokens.AccessToken, err = s.tokenManager.GenerateAccessToken(user.ID)
+	if err != nil {
+		http.Error(w, "can't generate access token: "+err.Error(), http.StatusUnauthorized)
+		return
+	}
+	tokens.RefreshToken, err = s.tokenManager.GenerateRefreshToken()
+	if err != nil {
+		http.Error(w, "can't generate refresh token: "+err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	fmt.Fprint(w, tokens.AccessToken, tokens.RefreshToken)
+}
+
+// SignUp implements the user registration process.
+func (s *Server) SignUp(w http.ResponseWriter, r *http.Request) {
+	var request domain.AuthRequest
+
+	err := json.NewDecoder(r.Body).Decode(&request)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+	defer r.Body.Close()
+
+	if err = validation.ValidateUserCredentials(request.Email, request.Password); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
```

Comment:
This is a bit misleading, it seems like you're actually checking if the credentials provided are correct. Let's rather validate the "Sign Up Request"